src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -978,53 +979,63 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                     lvalString = lval?.ToString() ?? string.Empty;
                 }
 
-                return ReplaceOperatorImpl(context, lvalString, rr, substitute);
+                return replacer.Replace(lvalString);
             }
             else
             {
                 List<object> resultList = new List<object>();
                 while (ParserOps.MoveNext(context, errorPosition, list))
                 {
                     string lvalString = PSObject.ToStringParser(context, ParserOps.Current(errorPosition, list));
-                    resultList.Add(ReplaceOperatorImpl(context, lvalString, rr, substitute));
+                    resultList.Add(replacer.Replace(lvalString));
                 }
 
                 return resultList.ToArray();
             }
         }
 
-        /// <summary>
-        /// ReplaceOperator implementation.
-        /// Abstracts away conversion of the optional substitute parameter to either a string or a MatchEvaluator delegate
-        /// and finally returns the result of the final Regex.Replace operation.
-        /// </summary>
-        /// <param name="context">The execution context in which to evaluate the expression.</param>
-        /// <param name="input">The input string.</param>
-        /// <param name="regex">A Regex instance.</param>
-        /// <param name="substitute">The substitute value.</param>
-        /// <returns>The result of the regex.Replace operation.</returns>
-        private static object ReplaceOperatorImpl(ExecutionContext context, string input, Regex regex, object substitute)
+        private struct ReplaceOperator
```

Comment:
Turns out `ReplaceOperator` is already the name of a static method. I guess let's just call it `ReplaceOperatorImpl` then :)

Reply:
:-))) Sorry, I'm already confused about the names.

Fixed.

Reply:
I've never liked `Impl` because it widens the "least representational gap", but I totally acknowledge it's fine in this case. Just want to make the case that I think we should be removing `Impl`s from the code in general.

Reply:
> I've never liked `Impl`

I tried to avoid it but .... :-)

Reply:
Maybe ReplaceOperatorProcessor?

Reply:
No need to change it! You've done great work here @iSazonov -- thanks!